module-4/backend/week_one.md

```diff
@@ -5,17 +5,34 @@ Fork this respository. Answer the questions to the best of your ability. Try to
 Note: When you're done, submit a PR. 
 
 1. What's the most useful thing you learned from completing the intermission week work?
+* I'm not sure. Just doing the sorting suite in JS helped with solving problems in a different language.
 2. What are some tools to help debug JavaScript code?
+* I can use console logging, debugger, and chrome dev tools.
 3. What are some tools you need in order to unit test your JavaScript?
+* I need to use Mocha as a test runner. I use Chai as an assertion library.
 4. What is the syntax for invoking a function?
+* functionName() ; Using the parenthesis invokes the function
 5. What's the difference between `==` and `===` in JavaScript?
+* === checks for type equality as well.
 6. What's the difference between asynchronous and synchronous JavaScript? 
+* Async JS allows data to be returned at another point in time, without waiting for the response to come back. This means we
+can run other code at the same time.
 7. What's a callback function and what are some reasons when we use/need callback functions?
+* Callback functions are called as a second argument in a function. As an example, we write a function that attaches an event
+handler to a DOM element. That event handler then triggers a callback function.
 8. What's the biggest difference between a promise and a callback?
+* A promise executes code when the DOM element or the data is ready.
 9. How do we setup a route when creating an API with Node and Express?
+* I have not looked at creating a route in express.
```

Comment:
yup - these are the old questions! next time, make sure you pull the latest version of master before you answer the questions 👍